drivers/volume/portworx/portworx.go

```diff
@@ -3946,6 +3955,113 @@ func (p *portworx) CleanupRestoreResources(*storkapi.ApplicationRestore) error {
 	return nil
 }
 
+// GetPodPatches returns driver-specific json patches to mutate the pod in a webhook
+func (p *portworx) GetPodPatches(podNamespace string, pod *v1.Pod) ([]k8sutils.JSONPatchOp, error) {
+	return p.getVirtLauncherPatches(podNamespace, pod)
+}
+
+func (p *portworx) getVirtLauncherPatches(podNamespace string, pod *v1.Pod) ([]k8sutils.JSONPatchOp, error) {
+	if pod.Labels["kubevirt.io"] != "virt-launcher" {
+		return nil, nil
+	}
+
+	// add the configMap as a volume to the pod
+	pod.Spec.Volumes = append(pod.Spec.Volumes, v1.Volume{
+		Name: statfsVolName,
+		VolumeSource: v1.VolumeSource{
+			ConfigMap: &v1.ConfigMapVolumeSource{
+				LocalObjectReference: v1.LocalObjectReference{
+					Name: statfsConfigMapName,
+				},
+			},
+		},
+	})
+
+	// add the configMap volume mount to all containers in the pod
+	for i := 0; i < len(pod.Spec.Containers); i++ {
+		pod.Spec.Containers[i].VolumeMounts = append(pod.Spec.Containers[i].VolumeMounts, v1.VolumeMount{
+			Name:      statfsVolName,
+			MountPath: path.Join(statfsSODirInVirtLauncher, statfsSOName),
+			SubPath:   statfsSOName,
+		})
+
+		pod.Spec.Containers[i].VolumeMounts = append(pod.Spec.Containers[i].VolumeMounts, v1.VolumeMount{
+			Name:      statfsVolName,
+			MountPath: "/etc/ld.so.preload",
+			SubPath:   "ld.so.preload",
+		})
+	}
+
+	containersBytes, err := json.Marshal(&pod.Spec.Containers)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal virt-launcher containers: %w", err)
+	}
+
+	volumesBytes, err := json.Marshal(&pod.Spec.Volumes)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal virt-launcher volumes: %w", err)
+	}
+
+	patches := []k8sutils.JSONPatchOp{
+		{
+			Operation: "add",
+			Path:      "/metadata/labels/portworx.io~1statfs", // ~1 encodes / in the json patch
+			Value:     []byte(`"true"`),
+		},
+		{
+			Operation: "replace",
+			Path:      "/spec/containers",
+			Value:     containersBytes,
+		},
+		{
+			Operation: "replace",
+			Path:      "/spec/volumes",
+			Value:     volumesBytes,
+		},
+	}
+
+	if err := p.createStatfsConfigMap(podNamespace); err != nil {
+		return nil, fmt.Errorf("failed to create config map: %w", err)
+	}
+
+	podName := pod.Name
+	if podName == "" {
+		podName = pod.GenerateName + "***"
+	}
+	logrus.Infof("Mutating kubevirt's virt-launcher pod %s/%s to support live migration of VM", podNamespace, podName)
+	return patches, nil
+}
+
+func (p *portworx) createStatfsConfigMap(cmNamespace string) error {
+	soPath := path.Join(statfsSODirInStork, statfsSOName)
+	statfsSOContents, err := ioutil.ReadFile(soPath)
+	if err != nil {
+		logrus.Errorf("Failed to read %s: %v", soPath, err)
+		return err
+	}
+
+	cm := &v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      statfsConfigMapName,
+			Namespace: cmNamespace,
+		},
+		BinaryData: map[string][]byte{
+			statfsSOName: statfsSOContents,
+		},
+		Data: map[string]string{
+			"ld.so.preload": path.Join(statfsSODirInVirtLauncher, statfsSOName),
```

Comment:
Where is the px_statfs.so getting injected into ld.so.preload?

Reply:
We add this configMap as a volume to the pod and inject the 2 keys in this configMap as files into the container's `/etc` dir. The configmap key becomes the name of the file, and value becomes the content.

See: https://docs.google.com/document/d/1HR3U3KWgite71jj3o5t44gvgiCG-Pc2veWzdWXbKtqU/edit